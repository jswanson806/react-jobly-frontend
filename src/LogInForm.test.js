import React from "react";
import {act, render, screen, fireEvent} from '@testing-library/react';
import SignInForm from './LogInForm.js';
import AppContext from "./helpers/AppContext.js";
import { MemoryRouter } from "react-router-dom";

import '@testing-library/jest-dom';


const logIn = jest.fn();

describe("login form smoke and snapshot tests", () => {

    it("should render without crashing", async () => {

        await act(async () => {render(
            <AppContext.Provider value={{logIn}}>
                <MemoryRouter>
                    <SignInForm />
                </MemoryRouter>
            </AppContext.Provider>
        )
        });
    });

    it("should match snapshot", async() => {
        let asFragment;

        await act(async () => {const {asFragment: fragment} = render(
            <AppContext.Provider value={{logIn}}>
                <MemoryRouter>
                    <SignInForm />
                </MemoryRouter>
            </AppContext.Provider>
        )
        asFragment = fragment;
        });

        expect(asFragment()).toMatchSnapshot();
    })
})

describe("Handles form input correctly", () => {

    it('updates search input correctly', async () => {
        await act(async () => {render(
            <AppContext.Provider value={{logIn}}>
                <MemoryRouter>
                    <SignInForm />
                </MemoryRouter>
            </AppContext.Provider>
        )
        });

        const userNameInput = screen.getByTestId("Login-form-username-input");
        const passwordInput = screen.getByTestId("Login-form-password-input");
        fireEvent.change(userNameInput, {target: {value: "testName"}});
        fireEvent.change(passwordInput, {target: {value: "password"}});
        expect(userNameInput).toHaveValue("testName");
        expect(passwordInput).toHaveValue("password");

    })

    it('should handle form submission', async () => {
        await act(async () => {render(
            <AppContext.Provider value={{logIn}}>
                <MemoryRouter>
                    <SignInForm />
                </MemoryRouter>
            </AppContext.Provider>
        )
        });

        const userNameInput = screen.getByTestId("Login-form-username-input");
        const passwordInput = screen.getByTestId("Login-form-password-input");
        fireEvent.change(userNameInput, {target: {value: "testName"}});
        fireEvent.change(passwordInput, {target: {value: "password"}});

        const btn = screen.getByTestId("Login-form-button");

        await act(async () => {
            fireEvent.click(btn);
        })

        expect(logIn).toHaveBeenCalledWith({ 
            username: "testName", 
            password: "password" 
        })
    })

})