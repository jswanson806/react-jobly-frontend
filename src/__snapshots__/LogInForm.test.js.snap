// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`login form smoke and snapshot tests should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="Login-card"
  >
    <h1
      class="Login-title"
    >
      Login
    </h1>
    <div>
      <form
        class="Login-form"
        data-testid="Login-form"
      >
        <label
          class="Login-label"
          for="username"
        >
          Username
        </label>
        <input
          class="Login-input"
          data-testid="Login-form-username-input"
          id="username"
          name="username"
          type="text"
          value=""
        />
        <label
          class="Login-label"
          for="password"
        >
          Password
        </label>
        <input
          class="Login-input"
          data-testid="Login-form-password-input"
          id="password"
          name="password"
          type="password"
          value=""
        />
        <button
          class="Login-button"
          data-testid="Login-form-button"
          type="submit"
        >
          Login
        </button>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
